builder/limitspec/limit_spec.go

```diff
@@ -8,7 +8,7 @@ import (
 )
 
 type Base struct {
-	Typ builder.ComponentType
+	Typ builder.ComponentType `json:"type,omitempty"`
```

Comment:
Yep. We could rename it to Type since it's uppercased/Exported. OTOH, since it's "typ" lowercased/private, maybe it's better to keep them even for the sake of readability ? WDYT?

Reply:
Not to hijack this PR which looks good, I'll merge it (let's discuss Type vs Typ on slack and open a follow up issue if needed)